optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorCreator.java

```diff
@@ -0,0 +1,265 @@
+            Optional<MethodDescriptor> setterDescriptor,
+            BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        if (!visitedMethods.contains(methodInfo)) {
+            transformers.produce(new BytecodeTransformerBuildItem(classInfo.name().toString(),
+                    (className, classVisitor) -> new OptaPlannerMethodEnhancingClassVisitor(classInfo, classVisitor, methodInfo,
+                            name, setterDescriptor)));
+            visitedMethods.add(methodInfo);
+        }
+        return setterDescriptor.map(md -> MethodDescriptor
+                .ofMethod(classInfo.name().toString(), getVirtualSetterName(name),
+                        md.getReturnType(), md.getParameterTypes()));
+    }
+
+    public static String getVirtualGetterName(String name) {
+        return "$get$optaplanner$__" + name;
+    }
+
+    public static String getVirtualSetterName(String name) {
+        return "$set$optaplanner$__" + name;
+    }
+
+    public static void generateFieldAccessor(AnnotationInstance annotationInstance, IndexView indexView,
+            ClassOutput classOutput, ClassInfo classInfo,
+            FieldInfo fieldInfo, BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        String generatedClassName = classInfo.name().prefix().toString() + ".$optaplanner$__"
+                + classInfo.name().withoutPackagePrefix() + "$__" + fieldInfo.name();
+        ClassCreator classCreator = ClassCreator
+                .builder()
+                .className(generatedClassName)
+                .interfaces(MemberAccessor.class)
+                .classOutput(classOutput)
+                .build();
+
+        GizmoMemberAccessor.MemberDescriptor member;
+        if (Modifier.isPublic(fieldInfo.flags())) {
+            member = new GizmoMemberAccessor.MemberDescriptor(fieldInfo, indexView);
+        } else {
+            addVirtualFieldGetter(classInfo, fieldInfo, transformers);
+            String methodName = getVirtualGetterName(fieldInfo.name());
+            MethodDescriptor setterDescriptor = MethodDescriptor.ofMethod(fieldInfo.declaringClass().name().toString(),
+                    getVirtualSetterName(fieldInfo.name()),
+                    "void",
+                    fieldInfo.type().name().toString());
+            MethodInfo methodInfo = MethodInfo.create(classInfo, methodName, new org.jboss.jandex.Type[] {}, fieldInfo.type(),
+                    (short) Modifier.PUBLIC);
+            member = new GizmoMemberAccessor.MemberDescriptor(fieldInfo, methodInfo, Optional.of(setterDescriptor), indexView,
+                    fieldInfo.name());
+        }
+        try {
+            GizmoMemberAccessor.defineAccessorFor(classCreator, member,
+                    (Class<? extends Annotation>) Class.forName(annotationInstance.name().toString()));
+            classCreator.close();
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException(e);
```

Comment:
When chaining an exception, always add a message and then the cause. The message should indicate, at the very least, what action is failing. So here that owuld be something like `"Unable to define assessor for member ("+ member + ").", e)`. That mentioning of the member's value is very handy when this occurs in a production log.